### PR TITLE
Promisify

### PR DESCRIPTION
**Pull request by jtbandes**
Several methods had both callback and async versions. Remove all callback-style methods in favor of returning Promises.

### CODE REVIEW

File: src/node/index.ts [hunk 1]
```diff
@@ -8,65 +8,51 @@
...
 import BagReader from "../BagReader";
-import { Callback } from "../types";
+import { Filelike } from "../types";
 
 // reader using nodejs fs api
-export class Reader {
+export class Reader implements Filelike {
   _filename: string;
-  _fd: number | null | undefined;
+  _file?: fs.FileHandle;
```
Comment (jtbandes) [inline src/node/index.ts:20]: With this change, jest now dumps a bunch of these:

```
(node:42051) [DEP0137] DeprecationWarning: Closing a FileHandle object on garbage collection is deprecated. Please close FileHandle objects explicitly using FileHandle.prototype.close(). In the future, an error will be thrown if a file descriptor is closed during garbage collection.
(node:42051) Warning: Closing file descriptor 25 on garbage collection
(node:42051) Warning: Closing file descriptor 28 on garbage collection
...
```

Should we care about these? I suspect we were already leaking file handles with the numeric `fd`s, but now we are just getting an explicit warning?

Currently the Bag API doesn't provide a close().

File: src/web/index.ts [hunk 2]
```diff
@@ -24,20 +24,21 @@ export class Reader {
...
-      cb(null, Buffer.from(reader.result as ArrayBuffer));
-    };
-    reader.onerror = function () {
-      reader.onload = null;
-      reader.onerror = null;
-      cb(reader.error ?? new Error("Unknown FileReader error"));
-    };
-    reader.readAsArrayBuffer(this._blob.slice(offset, offset + length));
+  async read(offset: number, length: number): Promise<Buffer> {
+    return await new Promise((resolve, reject) => {
```
Comment (jtbandes) [inline src/web/index.ts:28]: There's a newer Blob method `.arrayBuffer()` that returns a promise. However, jsdom doesn't support it so it breaks our unit tests 🤦 https://github.com/jsdom/jsdom/issues/2555  Also only somewhat newer browsers support it – so perhaps this is a sign the world is not yet ready for cleaner code.

File: src/node/index.ts (hunk 1 above)
Comment (defunctzombie) [reply src/node/index.ts:20]: How are we suppose to close the file handle object then? There are no explicit destructors in javascript...

**Comment by defunctzombie** [comment]
LGTM

File: src/node/index.ts (hunk 1 above)
Comment (jtbandes) [reply src/node/index.ts:20]: We could provide a close method on bag and ensure that each test manually closes a bag when it is done. We'd also have to closing even after something throws an error in the error tests...

**Pull request merged**